src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/MigrateCaseTest.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.adoption.adoptioncase.event;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.ccd.sdk.ConfigBuilderImpl;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.Event;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.CaseData;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.State;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.UserRole;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertThrows;
+import static uk.gov.hmcts.reform.adoption.adoptioncase.event.MigrateCase.MIGRATE_CASE;
+
+@ExtendWith(MockitoExtension.class)
+public class MigrateCaseTest extends EventTest {
+
+    @InjectMocks
+    private MigrateCase migrateCase;
+
+    @Test
+    void shouldSetupMigrateCaseEvent() {
+        final ConfigBuilderImpl<CaseData, State, UserRole> configBuilder = createCaseDataConfigBuilder();
+        migrateCase.configure(configBuilder);
+
+        assertThat(getEventsFrom(configBuilder).values())
+            .extracting(Event::getId)
+            .contains(MIGRATE_CASE);
+    }
+
+    @Test
+    void shouldPerformMigrationWithGivenIdAndNullifyMigrationId() {
+        CaseDetails<CaseData, State> caseDetails = getCaseDetails("ADOP-log");
+
+        AboutToStartOrSubmitResponse<CaseData, State> resp = migrateCase.aboutToSubmit(caseDetails, null);
+
+        assertThat(resp.getData().getMigrationId()).isNullOrEmpty();
+    }
+
+    @Test
+    void shouldThrowExceptionWhenMigrationRunWithInvalidId() {
+        CaseDetails<CaseData, State> caseDetails = getCaseDetails("THIS-MIGRATION-ID-IS-NOT-VALID");
+        assertThrows("No migration mapped to THIS-MIGRATION-ID-IS-NOT-VALID", RuntimeException.class,
+                     () -> migrateCase.aboutToSubmit(caseDetails, null)
+        );
+    }
+
+    private CaseDetails<CaseData, State> getCaseDetails(String migrationId) {
```

Comment:
not convinced "getCaseDetails" is a good name since it also adds the migration id & id.